Chore(lambda-chaos): Adds docs for AWS Lambda chaos to update function memory and timeout

uditgaurav opened the pull request:
It will add the docs for lambda function chaos to update memory limit and timeout value for a certain chaos duration to check the function stability.

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

uditgaurav committed:
Chore(lambda-chaos): Adds docs for AWS Lambda chaos to update function memory and timeout

It will add the docs for lambda function chaos to update memory limit and timeout value for a certain chaos duration to check the function stability.

avaakash commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+      </tr>
+      <tr>
+        <td> REGION </td>
+        <td> The region name of the target instance</td>
```
```suggestion
        <td> The region name of the target lambda function</td>
```

avaakash commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+      </tr>
+      <tr>
+        <td> TOTAL_CHAOS_DURATION </td>
+        <td> The total time duration for chaos insertion (sec) </td>
```
```suggestion
        <td> The total time duration for chaos insertion in seconds </td>
```

avaakash commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+      </tr>
+      <tr>
+        <td> CHAOS_INTERVAL </td>
+        <td> The interval (in sec) between successive instance termination.</td>
```
```suggestion
        <td> The interval (in seconds) between successive instance termination.</td>
```
IMO for docs using seconds would be a better approach

avaakash commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+        env:
+        # provide the function name for memory limit chaos
+        - name: FUNCTION_NAME
+          value: 'chaos-function'
```
Since we do not provide any default value for `MEMORY_IN_MEGABYTES`, this configuration will not work. I guess we can remove this example.

avaakash commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+        env:
+        # provide the function name for timeout chaos
+        - name: FUNCTION_NAME
+          value: 'chaos-function'
```
Similarly in this as well, we do not have any default value for `FUNCTION_TIMEOUT`

avaakash commented on docs/chaos-engineering/chaos-faults/chaos-faults.md:
```diff
@@ -295,6 +295,16 @@ Following Platform Chaos faults are available:
     <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-instance-stop">ecs-instance-stop</a></td>
   </tr>
+  <tr>
+    <td>Lambda Update function Timeout</td>
```
```suggestion
    <td>Lambda Update Function Timeout</td>
```

avaakash commented on docs/chaos-engineering/chaos-faults/chaos-faults.md:
```diff
@@ -295,6 +295,16 @@ Following Platform Chaos faults are available:
+    <td><a href="/docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout">lambda-update-function-timeout</a></td>
+  </tr>
+  <tr>
+    <td>Lambda Update function Memory</td>
```
```suggestion
    <td>Lambda Update Function Memory</td>
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+## Uses
+
+<details>
+<summary>View the uses of the experiment</summary>
```
```suggestion
<summary>View the uses of the fault</summary>
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+<details>
+<summary>View the uses of the experiment</summary>
+<div>
+Hitting a memory litmus is a very common and frequent scenario we find with lambda functions that can slow down the service and impacts their delivery. Such scenarios can still occur despite whatever availability aids AWS provides or we determine.
```
```suggestion
Hitting a memory limit is a very common and frequent scenario we find with lambda functions that can slow down the service and impacts their delivery. Such scenarios can still occur despite whatever availability aids AWS provides or we determine.
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+<div>
+Hitting a memory litmus is a very common and frequent scenario we find with lambda functions that can slow down the service and impacts their delivery. Such scenarios can still occur despite whatever availability aids AWS provides or we determine.
+
+Running out of memory due to a smaller limit interrupts the flow of the given function. So this category of chaos experiment helps you to build immunity on the application undergoing any such scenarios.
```
```suggestion
Running out of memory due to a smaller limit interrupts the flow of the given function. So this category of chaos fault helps you to build immunity on the application undergoing any such scenarios.
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-memory.md:
```diff
@@ -0,0 +1,172 @@
+    </table>
+</details>
+
+## Experiment Examples
```
```suggestion
## Fault Examples
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+## Uses
+
+<details>
+<summary>View the uses of the experiment</summary>
```
```suggestion
<summary>View the uses of the fault</summary>
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+<div>
+Hitting a timeout is a very common and frequent scenario we find with lambda functions that can break the service and impacts their delivery. Such scenarios can still occur despite whatever availability aids AWS provides or we determine.
+
+Getting timeout errors interrupts the flow of the given function. So this category of chaos experiment helps you to build the immunity of the application undergoing any such scenarios.
```
```suggestion
Getting timeout errors interrupts the flow of the given function. So this category of chaos fault helps you to build the immunity of the application undergoing any such scenarios.
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+
+:::
+
+## Experiment Tunables
```
```suggestion
## Fault Tunables
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+## Experiment Tunables
+
+<details>
+    <summary>Check the Experiment Tunables</summary>
```
```suggestion
    <summary>Check the Fault Tunables</summary>
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+    </table>
+</details>
+
+## Experiment Examples
```
```suggestion
## Fault Examples
```

neelanjan00 commented on docs/chaos-engineering/chaos-faults/aws/lambda-update-function-timeout.md:
```diff
@@ -0,0 +1,171 @@
+
+### Common and AWS specific tunables
+
+Refer the [common attributes](../common-tunables-for-all-experiments) and [AWS specific tunable](./aws-experiments-tunables) to tune the common tunables for all experiments and aws specific tunables.
```
```suggestion
Refer the [common attributes](../common-tunables-for-all-experiments) and [AWS specific tunable](./aws-experiments-tunables) to tune the common tunables for all faults and aws specific tunables.
```

SudheendraKatte committed:
SRM docs migration

uditgaurav committed:
Fixes the typo, remove some fault examples and update experiment to fault

Update the experiment references to fault in all the experiments and fix typo check sanity for the lambda experiments.

SudheendraKatte committed:
Incorporated review comments for PR139

SudheendraKatte committed:
Incorporated review comments

KimberlyFields committed:
Merge pull request #139 from harness/srm-docs-migration

SRM Docs Migration

uditgaurav committed:
Merge branch 'harness:main' into docs_lambda_update_func_chaos_

uditgaurav committed:
modify experiment to fault

uditgaurav commented:
Closing this PR as it is added here - https://github.com/harness/developer-hub/pull/143
